lsp/Main.hs

```diff
@@ -1,45 +1,151 @@
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE LambdaCase #-}
 module Main where
-  
+
 import Language.LSP.Server
 import Language.LSP.Types
 import Control.Monad.IO.Class
 import qualified Data.Text as T
+import Lexer
+import Data.List (find)
+import Data.Text.IO (hPutStrLn)
+import System.IO (stderr)
+
+import qualified Language.LSP.Types            as J
+import           Language.LSP.Diagnostics
+import Data.Traversable
+
+type Config = ()
 
 handlers :: Handlers (LspM ())
 handlers = mconcat
   [ notificationHandler SInitialized $ \_not -> do
-      let params = ShowMessageRequestParams MtInfo "Turn on code lenses?"
-            (Just [MessageActionItem "Turn on", MessageActionItem "Don't"])
-      _ <- sendRequest SWindowShowMessageRequest params $ \res ->
-        case res of
-          Right (Just (MessageActionItem "Turn on")) -> do
-            let regOpts = CodeLensRegistrationOptions Nothing Nothing (Just False)
-              
-            _ <- registerCapability STextDocumentCodeLens regOpts $ \_req responder -> do
-              let cmd = Command "Say hello" "lsp-hello-command" Nothing
-                  rsp = List [CodeLens (mkRange 0 0 0 100) (Just cmd) Nothing]
-              responder (Right rsp)
-            pure ()
-          Right _ ->
-            sendNotification SWindowShowMessage (ShowMessageParams MtInfo "Not turning on code lenses")
-          Left err ->
-            sendNotification SWindowShowMessage (ShowMessageParams MtError $ "Something went wrong!\n" <> T.pack (show err))
+      -- let params = ShowMessageRequestParams MtInfo "Turn on code lenses?"
+      --       (Just [MessageActionItem "Turn on", MessageActionItem "Don't"])
+      -- _ <- sendRequest SWindowShowMessageRequest params $ \case
+      --     Right (Just (MessageActionItem "Turn on")) -> do
+      --       let regOpts = CodeLensRegistrationOptions Nothing Nothing (Just False)
+
+      --       _ <- registerCapability STextDocumentCodeLens regOpts $ \_req responder -> do
+      --         let cmd = Command "Say hello" "lsp-hello-command" Nothing
+      --             rsp = List [CodeLens (mkRange 0 0 0 100) (Just cmd) Nothing]
+      --         responder (Right rsp)
+      --       pure ()
+      --     Right _ ->
+      --       sendNotification SWindowShowMessage (ShowMessageParams MtInfo "Not turning on code lenses")
+      --     Left err ->
+      --       sendNotification SWindowShowMessage (ShowMessageParams MtError $ "Something went wrong!\n" <> T.pack (show err))
```

Comment:
This commented-out block I think we can just remove.
I know where to get it :-)

Reply:
Good point